### PR TITLE
Remove ariaLabel prop and use same translation close for dialog

### PR DESCRIPTION
**Pull request by bryancunningham-okta**
[OKTA-683644](https://oktainc.atlassian.net/browse/OKTA-683644)

## Testing & Screenshots

- [ ] I have confirmed this change with my designer and the Odyssey Design Team.

### CODE REVIEW

**Review by chrispulsinelli-okta — APPROVED** [review]

File: packages/odyssey-react-mui/src/Dialog.tsx [hunk 1]
```diff
@@ -111,7 +116,7 @@ const Dialog = ({
       <DialogTitle translate={translate}>
         {title}
         <Button
-          ariaLabel={ariaLabel}
+          ariaLabel={t("close.text")}
```
Comment (chrispulsinelli-okta) [inline packages/odyssey-react-mui/src/Dialog.tsx:119]: Do we need to fall back to the supplied one for now?
Also, if someone uses this prop, can we give them a warning / error at build time?

Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/Dialog.tsx:119]: This translation already exists for `Toast` so, I just used the same one. Not sure about warning at build time. No strong feelings either way

Comment (chrispulsinelli-okta) [reply packages/odyssey-react-mui/src/Dialog.tsx:119]: I just wonder if we should warn users about this, if we detect they set the prop. We're pretty much dropping the value on the floor now.

**Review by chrispulsinelli-okta — DISMISSED** [review]
I'm good with the changes and leave it up to you if you want to leave a warning for users if they set the prop we deprecated.

File: packages/odyssey-react-mui/src/Dialog.tsx (hunk 1 above)
Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/Dialog.tsx:119]: @chrispulsinelli-okta I'm fine with warning users, if that is what we want to do. How do I best accomplish that?

File: packages/odyssey-react-mui/src/Dialog.tsx [hunk 2]
```diff
@@ -31,6 +32,11 @@ import {
 import type { AllowedProps } from "./AllowedProps";
 
 export type DialogProps = {
+  /**
+   * @deprecated `aria-label` for close button comes from translation file
+   */
+  /** **Deprecated:** `aria-label` for close button comes from translation file */
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Dialog.tsx:38]: We can remove this line:
```suggestion
```

**Review by KevinGhadyani-Okta — COMMENTED** [review]
Other than one comment removal, this is approved 👍

**Review by KevinGhadyani-Okta — APPROVED** [review]

**Pull request merged**